detection_rules/rule.py

```diff
@@ -269,6 +269,29 @@ class EQLRuleData(QueryRuleData):
     type: Literal["eql"]
     language: Literal["eql"]
 
+    @cached_property
+    def max_span(self) -> Optional[eql.ast.TimeRange]:
+        """Maxspan value for sequence rules if defined."""
+        if eql.utils.get_query_type(self.ast) == 'sequence' and hasattr(self.ast.first, 'max_span'):
+            return self.ast.first.max_span
+
+    @cached_property
+    def look_back(self) -> Union[eql.ast.TimeRange, Literal['unknown']]:
+        """Lookback value of a rule."""
+        # https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#date-math
+        from_ = self.from_ or 'now-6m'
+        from_ = from_.split('-')
+        to = self.to
+
+        if len(from_) != 2 or from_[0] != 'now' or to not in (None, 'now'):
+            return 'unknown'
+
+        _, lookback = from_
+        quantity = int(lookback[-2])
+        unit = eql.ast.TimeUnit(lookback[-1])
+
+        return eql.ast.TimeRange(quantity, unit)
```

Comment:
this logic is tough to read and check. I think avoiding split and explicit indexes would help.

personally, i like this approach:
```python
    @staticmethod
    def convert_es_time():
        if lookback.startswith("now"):
            lookback = lookback[len("now"):]
            sign = lookback[0]  # + or -
            span = lookback[1:]
            amount = int("".join(char for char in lookback if char.isdigit()))
            unit = "".join(char for char in lookback if char.isalpha())
            return amount * eql.ast.TimeSpan(unit).as_milliseconds() * (-1 if sign == "-" else 1)
            
    def convert_es_time(self) -> int:
        return self.convert_relative_delta(self.to or "now") - self.convert_relative_delta(self.from_or "now-6m")
```

Reply:
I implemented a variation of this